src/main/java/com/microsoft/sqlserver/jdbc/dtv.java

```diff
@@ -861,10 +863,28 @@ private void sendTemporal(DTV dtv, JavaType javaType, Object value) throws SQLSe
                                             timestampNormalizedCalendar(calendar, javaType, conn.baseYear()),
                                             subSecondNanos, (valueLength), isOutParam, statement);
                                 }
-                            } else
-                                tdsWriter.writeRPCDateTime2(name,
-                                        timestampNormalizedCalendar(calendar, javaType, conn.baseYear()),
-                                        subSecondNanos, TDS.MAX_FRACTIONAL_SECONDS_SCALE, isOutParam);
+                            } else {
+                                if (jdbcType == JDBCType.SMALLDATETIME) {
```

Comment:
this whole block seems very awkward, we're in a switch for these types and then we're checking the types again multiple times

Reply:
also, I know this wasn't added in this PR, but the check for isKatmaiOrLater is obselete as everything supported is > Katmai now so this should be removed since we will never be testing the else case

Reply:
I divided the cases into each switch case. I'm guessing in the past all 3 date/time types used the same `tdsWriter` call and it just got messy over time.